### PR TITLE
BENCH-161 Refactored codebase

### PR DESCRIPTION
**Pull request by martintaylor1635**
- Refactored codebase to use product rather tham item.

Apologies in advance for the size of the PR, also **high risk** that some mistakes have been made
so it would be good to merge this once at least 3 approvals have been made in my opinion.

### CODE REVIEW

**Comment by MohammadIqbalAD** [comment]
There are some mentions of Item inside the Target folder
image

**Comment by martintaylor1635** [comment]
> There are some mentions of Item inside the Target folder image

Hey! Can you please try and re-run `mvn clean install`

File: README.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Answer King REST API
 
-The Answer King REST API is the backend for a fast food business. It allows items and categories to be created and 
-orders to be placed. An item can be included in multiple categories.
+The Answer King REST API is the backend for a fast food business. It allows products and categories to be created and 
+orders to be placed. An product can be included in multiple categories.
```
Comment (AndyFlintAnswerDigital) [inline README.md:4]: 'An product' should be 'A product'

Comment (martintaylor1635) [reply README.md]: Fixed!

**Comment by MohammadIqbalAD** [comment]
> > There are some mentions of Item inside the Target folder image
> 
> Hey! Can you please try and re-run `mvn clean install`

Yep that solved it for me. Oops!

File: src/main/java/com/answerdigital/academy/answerking/service/OrderService.java [hunk 2]
```diff
@@ -56,60 +56,64 @@ public Order updateOrder(final Long orderId, final OrderRequest orderRequest) {
...
+                .filter(lineItem -> lineItem.getProduct() == product)
                 .findFirst();
 
-        if (existingOrderItem.isEmpty()) {
-            throw new NotFoundException(String.format("Item id = %s is not in the basket of order id = %s", orderId, itemId));
+        if (existingLineItem.isEmpty()) {
+            throw new NotFoundException(
+                    String.format("Product id = %d is not in the basket of order id = %d", orderId, productId)
+            );
         }
```
Comment (MohammadIqbalAD) [inline src/main/java/com/answerdigital/academy/answerking/service/OrderService.java:95]: Was changing '%d' to '%s' intentional? If so, what's the difference?

Comment (martintaylor1635) [reply src/main/java/com/answerdigital/academy/answerking/service/OrderService.java:95]: Hi @MohammadIqbalAD - I changed it to %d from %s as %s is shorthand for any type (more specifically though used for Strings) and %d is used for integer types (incl. byte, short, int, long, bigint). As orderId and itemId are both numeric, I opted for %d.

Comment (MohammadIqbalAD) [reply src/main/java/com/answerdigital/academy/answerking/service/OrderService.java:95]: Ah makes sense now. Thanks @martintaylor1635

**Review by AndyFlintAnswerDigital — APPROVED** [review]

**Review by MohammadIqbalAD — APPROVED** [review]

**Pull request merged**